### PR TITLE
Judith Pillado 

### PR DESCRIPTION
**Pull request by judithpillado**

### CODE REVIEW

**Comment by KathleenScriver** [comment]
Great job with all these commits.

File: lib/cookbook.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @recipes << recipe
+  end
+
+  def ingredients
+    all_ingredients = @recipes.flat_map do |recipe|
+      recipe.ingredients
+    end
+    all_ingredients.map do |ingredient|
+      ingredient.name
+    end.uniq
```
Comment (KathleenScriver) [inline lib/cookbook.rb:18]: Great job. I appreciate how you broke this up into two parts. I think the argument would be that you are now using additional storage by creating the `all_ingredients` variable, but I like this readability.

File: lib/cookbook.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def highest_calorie_meal
+    @recipes.max_by do |recipe|
```
Comment (KathleenScriver) [inline lib/cookbook.rb:22]: Great use of max_by here.

File: lib/pantry.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+class Pantry
+  attr_reader :stock
+
+  def initialize
+    @stock = Hash.new(0)
```
Comment (KathleenScriver) [inline lib/pantry.rb:5]: Great job accounting for the default value here.

File: lib/pantry.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def stock_check(ingredient)
+    @stock[ingredient]
+  end
+
+  def restock(ingredient, amount)
+    @stock[ingredient] += amount
```
Comment (KathleenScriver) [inline lib/pantry.rb:13]: Great job!

File: lib/pantry.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def enough_ingredients_for?(recipe)
+    recipe.ingredients_required.all? do |ingredient, amount|
+      if stock_check(ingredient) >= amount
```
Comment (KathleenScriver) [inline lib/pantry.rb:18]: Great job here! However, with `all?` it will evaluate the block to truth or falsey so you don't even need the if statement here!

File: lib/recipe.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class Recipe
+  attr_reader :name, :ingredients_required
+
+  def initialize(name)
+    @name = name
+    @ingredients_required = Hash.new(0)
```
Comment (KathleenScriver) [inline lib/recipe.rb:6]: Great job here.

File: lib/recipe.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def add_ingredient(ingredient, amount)
+    @ingredients_required[ingredient] += amount
+  end
+
+  def ingredients
+    @ingredients_required.keys
```
Comment (KathleenScriver) [inline lib/recipe.rb:14]: Great!

File: lib/recipe.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def total_calories
+    @ingredients_required.sum do |ingredient, amount|
```
Comment (KathleenScriver) [inline lib/recipe.rb:18]: Great use of #sum.

File: test/cookbook_test.rb [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+  end
+
+  def test_it_exists
+    assert_instance_of CookBook, @cookbook
+  end
+
+  def test_add_recipe
+    @cookbook.add_recipe(@recipe1)
+    @cookbook.add_recipe(@recipe2)
+    assert_equal [@recipe1, @recipe2], @cookbook.recipes
```
Comment (KathleenScriver) [inline test/cookbook_test.rb:39]: I would also like to see the initial state of #recipes tested so that we know it starts with an empty array.

File: test/cookbook_test.rb [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    @cookbook.add_recipe(@recipe2)
+    @pantry.restock(@ingredient1, 5)
+    @pantry.restock(@ingredient1, 10)
+    assert_equal false, @pantry.enough_ingredients_for?(@recipe1)
+
+    @pantry.restock(@ingredient2, 7)
+    assert_equal false, @pantry.enough_ingredients_for?(@recipe1)
+
+    @pantry.restock(@ingredient2, 1)
+    assert_equal true, @pantry.enough_ingredients_for?(@recipe1)
```
Comment (KathleenScriver) [inline test/cookbook_test.rb:65]: I appreciate that you tested for the true case and the false case.

File: test/pantry_test.rb [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+  def test_stock
+    pantry = Pantry.new
+    assert_equal ({}), pantry.stock
+  end
+
+  def test_stock_check
+    pantry = Pantry.new
+    ingredient1 = Ingredient.new({name: "Cheese", unit: "oz", calories: 50})
+
+    assert_equal 0, pantry.stock_check(ingredient1)
```
Comment (KathleenScriver) [inline test/pantry_test.rb:22]: Great job testing for the default value here.
